test/src/test/java/hudson/model/AutoCompletionCandidatesTest.java

```diff
@@ -60,6 +59,32 @@ public void completion() throws Exception {
         assertContains(c, "../bar", "../foo");
     }
 
+    @Test
+    public void completion2() throws Exception {
+        FreeStyleProject foo = j.createFreeStyleProject("foo");
+        MatrixProject bar = j.jenkins.createProject(MatrixProject.class, "bar");
+        bar.setAxes(new AxisList(new TextAxis("x","1","2","3")));
+        MatrixConfiguration x3 = bar.getItem("x=3");
+
+        AutoCompletionCandidates c;
+
+        c = AutoCompletionCandidates.ofJobNames("", j.jenkins);
+        assertContains(c, "foo", "bar", "bar/x=1", "bar/x=2", "bar/x=3");
+
+        c = AutoCompletionCandidates.ofJobNames("ba", j.jenkins);
+        assertContains(c,  "bar", "bar/x=1", "bar/x=2", "bar/x=3");
+
+        c = AutoCompletionCandidates.ofJobNames("bar/", j.jenkins);
+        assertContains(c, "bar/x=1", "bar/x=2", "bar/x=3");
+
+        c = AutoCompletionCandidates.ofJobNames( "", x3.getParent());
+        assertContains(c, "../foo", "../bar", "x=1", "x=2", "x=3");
+
+        // relative path
+        c = AutoCompletionCandidates.ofJobNames(Item.class, "../", x3, x3.getParent());
```

Comment:
This is not testing the new method for the relative path and I'm not confident that the new method is working properly for the relative paths. Could you please use the new method signature to validate the relative path? Only the new one is used in the `doAutoComplete*` method, so we might have a feature regression on the form.

Reply:
You are right. That part is clearly my mistake. I will fix that part of the test. I got a comment that ```getAllItems()``` is expensive to run but I really need to use it. I am not experienced enough to do something  more complex than that.